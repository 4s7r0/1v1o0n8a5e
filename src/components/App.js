import React, { Fragment, useState, useEffect } from 'react'
import { NavLink, Route } from 'react-router-dom'
import Sidebar from 'react-sidebar'
import styled, { createGlobalStyle } from 'styled-components'
import { color, width } from 'styled-system'
import theme from '../settings/theme'
import routes from '../pages/routes'
import resetStyles from '../settings/reset.styles'
import globalFonts from '../settings/global.fonts'
import globalStyles from '../settings/global.styles'
import useWindowSize from '../hooks/useWindowSize'
import { Grommet, Box } from 'grommet'

const GlobalStyle = createGlobalStyle`
  ${resetStyles}
  ${globalStyles}
  ${globalFonts}
`

const AppWrapper = styled.div`
  min-height: 100vh;
  padding: 0;
  margin: 0;
  ${width}
  ${color}
`

const AppHeader = (props) => (
  <Box
    tag='header'
    direction='row'
    align='start'
    justify='start'
    background='brand'
    pad={{ left: 'medium', right: 'small', vertical: 'small' }}
    elevation='medium'
    style={{ zIndex: '1' }}
    overflow='hidden'
    height='auto'
    minHeight='6rem'
    {...props}
  />
)

const AppBody = styled.div`
  margin-top: 20px;
  margin-left: 10vw;
  margin-right: 10vw;
  text-align: left;
  ${color}
`

const AppTitle = styled.div`
  display: inline-block;
  font-weight: 900;
  font-family: 'Aldrich', sans-serif;
  font-size: calc(2.5em + 2vmin);
`

const Button = styled.button`
  border: 0;
  font-size: calc(2.5em + 2vmin);
  line-height: 1.45rem;
  background: transparent;
  ${color}
`

const AppName = styled.input`
  font-family: 'Aldrich', sans-serif;
  font-size: 4rem;
  text-align: left;
  border: 0;
  color: #265C83;
  background-color: transparent;
`

const SidebarHeader = styled.div`
  padding: 34px;
  font-family: 'Aldrich', sans-serif;
  font-size: 1em;
  min-height: 10vh;
  ${color}
`

function useFormInput(initialValue) {
  const [value, setValue] = useState(initialValue)
  function handleChange(e) {
    setValue(e.target.value)
  }
  return {
    value,
    onChange: handleChange,
  }
}

function useDocumentTitle(title) {
  useEffect(() => {
    document.title = title
  })
}  

const markActive = (match, location) => {
  if (!match) {
    return false
  }
  console.log(JSON.stringify(location))
  return true
}

function App() {
  const [sidebarDocked, setSidebarDocked] = useState(false)
  const [open, setSidebarOpen] = useState(false)
  const name = useFormInput('1v1o0n8a5e')
  const size = useWindowSize()
  const nameWidth = (size.width - (size.width/2)) + 'px'
  useDocumentTitle(name.value)

  function onSetSidebarDocked(dock) {
    setSidebarDocked(dock)
  }
  
  function onSetSidebarOpen(opn) {
    setSidebarOpen(opn)
  }

  return(
    <Grommet theme={theme}>
      <AppWrapper className='App' color='#1b1b25' bg='#6d6d6d'>
        <GlobalStyle />
        <Sidebar
          sidebar={
            <div style={{
              textAlign: 'center', margin: '0', paddingTop: '1rem', fontFamily: 'Electrolize', fontSize: '1.5em', background: '#26262a', height: '100vh'
            }}>
              <SidebarHeader>
                v0.1.0
              </SidebarHeader>
              <ul>
                {routes.map(route => {
                  if (route.path === '/') {
                    return (<Fragment key={route.id}></Fragment>) }
                  else {
                    return (
                      <li key={route.id}>
                        <NavLink exact to={route.path} isActive={markActive} className='noselect' activeClassName='navselect'>
                          {route.sidebar}
                        </NavLink>
                      </li>
                    )
                  }
                })}
              </ul>
            </div>
          }
          styles={{ sidebar: { top: null, width: '10vw', minWidth: '185px', backgroundColor: '#gray', margin: '0', padding: '0' } }}
          docked={sidebarDocked}
          open={open}
          onSetOpen={onSetSidebarOpen}
          touch
        >
          <AppHeader>
            <AppTitle>
              <Button color='#265C83' onClick={() => onSetSidebarDocked(!sidebarDocked)}>
                <span className='logo' role='img' aria-label='Waxing cresent moon' style={{ paddingLeft: '20px', marginRight: '50px' }}>
                  ðŸŒ’
                </span>
              </Button>
              <AppName {...name} className='text-pulse' width={nameWidth} spellCheck='false' />
            </AppTitle>
          </AppHeader>
          <AppBody>
            {routes.map(route => (
              <Route key={route.id} path={route.path} exact={route.exact} component={route.main} />
            ))}
          </AppBody>
        </Sidebar>
      </AppWrapper>
    </Grommet>
  )
}
export default App
