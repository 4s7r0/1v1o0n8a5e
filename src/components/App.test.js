import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { BrowserRouter as Router } from 'react-router-dom'
import Sidebar from 'react-sidebar'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <Router>
      <App />
    </Router>, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders the sidebar', () => {
  // I have no fucking clue what to do here and there isn't a fucking website or video that helps.
})
