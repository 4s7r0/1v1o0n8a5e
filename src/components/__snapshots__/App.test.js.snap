// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads a theme file 1`] = `
Object {
  "global": Object {
    "colors": Object {
      "aqua": "#7FDBFF",
      "bg_aqua": "#7FDBFF",
      "bg_black": "#111111",
      "bg_blue": "#0074D9",
      "bg_fuchsia": "#F012BE",
      "bg_gray": "#AAAAAA",
      "bg_green": "#2ECC40",
      "bg_lime": "#01FF70",
      "bg_maroon": "#85144B",
      "bg_navy": "#001F3F",
      "bg_olive": "#3D9970",
      "bg_orange": "#FF851B",
      "bg_purple": "#B10DC9",
      "bg_red": "#FF4136",
      "bg_silver": "#DDDDDD",
      "bg_teal": "#39CCCC",
      "bg_white": "#FFFFFF",
      "bg_yellow": "#FFDC00",
      "black": "#111111",
      "blue": "#0074D9",
      "border_aqua": "#7FDBFF",
      "border_black": "#111111",
      "border_blue": "#0074D9",
      "border_fuchsia": "#F012BE",
      "border_gray": "#AAAAAA",
      "border_green": "#2ECC40",
      "border_lime": "#01FF70",
      "border_maroon": "#85144B",
      "border_navy": "#001F3F",
      "border_olive": "#3D9970",
      "border_orange": "#FF851B",
      "border_purple": "#B10DC9",
      "border_red": "#FF4136",
      "border_silver": "#DDDDDD",
      "border_teal": "#39CCCC",
      "border_white": "#FFFFFF",
      "border_yellow": "#FFDC00",
      "brand": "#1b1b25",
      "fill_aqua": "#7FDBFF",
      "fill_black": "#111111",
      "fill_blue": "#0074D9",
      "fill_fuchsia": "#F012BE",
      "fill_gray": "#AAAAAA",
      "fill_green": "#2ECC40",
      "fill_lime": "#01FF70",
      "fill_maroon": "#85144B",
      "fill_navy": "#001F3F",
      "fill_olive": "#3D9970",
      "fill_orange": "#FF851B",
      "fill_purple": "#B10DC9",
      "fill_red": "#FF4136",
      "fill_silver": "#DDDDDD",
      "fill_teal": "#39CCCC",
      "fill_white": "#FFFFFF",
      "fill_yellow": "#FFDC00",
      "fuchsia": "#F012BE",
      "gray": "#AAAAAA",
      "green": "#2ECC40",
      "lime": "#01FF70",
      "maroon": "#85144B",
      "navy": "#001F3F",
      "olive": "#3D9970",
      "orange": "#FF851B",
      "purple": "#B10DC9",
      "red": "#FF4136",
      "silver": "#DDDDDD",
      "stroke_aqua": "#7FDBFF",
      "stroke_black": "#111111",
      "stroke_blue": "#0074D9",
      "stroke_fuchsia": "#F012BE",
      "stroke_gray": "#AAAAAA",
      "stroke_green": "#2ECC40",
      "stroke_lime": "#01FF70",
      "stroke_maroon": "#85144B",
      "stroke_navy": "#001F3F",
      "stroke_olive": "#3D9970",
      "stroke_orange": "#FF851B",
      "stroke_purple": "#B10DC9",
      "stroke_red": "#FF4136",
      "stroke_silver": "#DDDDDD",
      "stroke_teal": "#39CCCC",
      "stroke_white": "#FFFFFF",
      "stroke_yellow": "#FFDC00",
      "teal": "#39CCCC",
      "white": "#FFFFFF",
      "yellow": "#FFDC00",
    },
  },
}
`;
